src/Mod/Start/Gui/NewFileButton.cpp

```diff
@@ -0,0 +1,86 @@
+ *                                                                          *
+ *   FreeCAD is distributed in the hope that it will be useful, but         *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU       *
+ *   Lesser General Public License for more details.                        *
+ *                                                                          *
+ *   You should have received a copy of the GNU Lesser General Public       *
+ *   License along with FreeCAD. If not, see                                *
+ *   <https://www.gnu.org/licenses/>.                                       *
+ *                                                                          *
+ ***************************************************************************/
+
+#include "NewFileButton.h"
+#include <QFont>
+#include <QIcon>
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <algorithm>
+
+namespace StartGui
+{
+
+NewFileButton::NewFileButton(const NewButton& newButton)
+{
+    auto hGrp = App::GetApplication().GetParameterGroupByPath(
+        "User parameter:BaseApp/Preferences/Mod/Start");
+    const auto newFileIconSize = static_cast<int>(hGrp->GetInt("NewFileIconSize", 48));  // NOLINT
+
+    auto mainLayout = gsl::owner<QHBoxLayout*>(new QHBoxLayout(this));
+    mainLayout->setAlignment(Qt::AlignVCenter);
+    auto iconLabel = gsl::owner<QLabel*>(new QLabel(this));
+    mainLayout->addWidget(iconLabel);
+    QIcon baseIcon(newButton.iconPath);
+    iconLabel->setPixmap(baseIcon.pixmap(newFileIconSize, newFileIconSize));
+    iconLabel->setPixmap(baseIcon.pixmap(newFileIconSize, newFileIconSize));
+
+    auto textLayout = gsl::owner<QVBoxLayout*>(new QVBoxLayout);
+    auto textLabelLine1 = gsl::owner<QLabel*>(new QLabel(this));
+    textLabelLine1->setText(newButton.heading);
+    QFont font = textLabelLine1->font();
+    font.setWeight(QFont::Bold);
+    textLabelLine1->setFont(font);
+    auto textLabelLine2 = gsl::owner<QLabel*>(new QLabel(this));
+    textLabelLine2->setText(newButton.description);
+    textLabelLine2->setWordWrap(true);
+    textLayout->addWidget(textLabelLine1);
+    textLayout->addWidget(textLabelLine2);
+    textLayout->setSpacing(0);
+    mainLayout->addItem(textLayout);
+    mainLayout->addStretch();
+    setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Expanding);
+}
+
+QSize NewFileButton::minimumSizeHint() const
+{
+    auto hGrp = App::GetApplication().GetParameterGroupByPath(
+        "User parameter:BaseApp/Preferences/Mod/Start");
+    const auto cardSpacing = static_cast<int>(hGrp->GetInt("FileCardSpacing", 25));       // NOLINT
+    const auto newFileIconSize = static_cast<int>(hGrp->GetInt("NewFileIconSize", 48));   // NOLINT
+    const auto cardLabelWith = static_cast<int>(hGrp->GetInt("FileCardLabelWith", 180));  // NOLINT
```

Comment:
Same comment with `// NOLINT` - if it is complaining about magic numbers let's just add proper constexpr with the defaults.

Reply:
Idk why there is even a parameter for this

Reply:
that's a good question, they react to
QWidget {margin-left: 50px; margin-top: 5px; width: 20em;}
so maybe adding an id attribute would be better, that way theme devs can more easily target this widgets